### ISSUE TITLE
Mark as verified: org.videolan.VLC.Plugin.pause_click

### ISSUE THREAD

**Issue by nurupo** [flathub/flathub#5117]
Please mark the org.videolan.VLC.Plugin.pause_click extension as verified.

![image](https://github.com/flathub/flathub/assets/1855294/b1461e9e-bdab-490d-a8cc-6917d4bfebd3)

I'm the author/developer/maintainer of https://github.com/nurupo/vlc-pause-click-plugin

I'm also the maintainer of the Flathub flatpak for it https://github.com/flathub/org.videolan.VLC.Plugin.pause_click

I'm unable to verify it via the Flathub website as it asks me to place a token at a specific page on videolan.org:

![image](https://github.com/flathub/flathub/assets/1855294/d49e82e2-5944-4f94-992a-0ea0fd93c534)

I do not control web pages on videolan.org, I'm just a developer of a 3rd party plugin for VLC. I appears like the verification process was not very well-thought-out, as the same verification process as for apps doesn't work very well for extensions.

**Comment by bbhtt** [comment]
The website shouldn't show a dedicated page for extensions or shouldn't show verified/unverified badges on them. 

The design of Flatpak means if you don't own the target application, you don't own the app-id of the extension, meaning you cannot verify them.

**Comment by nurupo** [comment]
>The design of Flatpak means if you don't own the target application, you don't own the app-id of the extension, meaning you cannot verify them.

I don't care about owning the app-id of the extension, nor do I claim to be a VLC developer (I am though a VLC contributor, not that it matters here). I only care about it saying "Verified" to mean that the extension is indeed published by the author of the extension, i.e. by me, as this seems to have been the intention of the Verified badge:

>[A verified app is an app that is published on Flathub by its original developer or a third party approved by the developer. Flathub has verified the developer's identity using the app ID and the developer's website or profile on a source code hosting site.](https://docs.flathub.org/docs/for-users/verification/)

Also, ignoring that extensions shouldn't have the verified badge for a moment, to say that because I don't own the org.videolan.* app-id namespace, you can't verify that I am indeed the original developer of the extension, is kind of absurd, it doesn't make sense to me. Who else can get the extension verified if not its original developer? Even the people who do own the org.videolan app-id namespace can't verify it, as they can't claim to be the original developers of my extension either, as it was developer solely by be, with a few small contributions from non-owners of the org.videolan app-id namespace.

>The website shouldn't show a dedicated page for extensions 

Why extensions should not have dedicated pages? I'm advocating for extensions to have dedicated pages!

I think it's a good idea to have dedicated pages for extensions, to have a place for extensions to provide descriptions of what they do, instructions on how to activate them and screenshots of them in the app, if applicable. This is very useful for my extension in particular, as after my extension is installed, it must be manually enabled in the VLC before it can be used, and it's not obvious where to enable the extension in the VLC (in fact, you must enable it in two places in the case of my extension, which makes it even less obvious), and the dedicated page allows me to show instructions on how to enable the extension, along with screenshots. (Flathub for some reason resizes the screenshots to be way too small to the point you can't even read the text, so they are not very useful, wish Flathub didn't do that). Without the dedicated page, users would have no clue how to activate the extension and would file GitHub issues / spam my email, saying that the extension doesn't work.

With extensions having dedicated pages, it also makes sense to have verification badges for them, but the current verification workflow is tailored more towards the app verification, it doesn't seem to account for extensions being developed by 3rd party developers. This is something that needs to be improved.

Since the verification process on the website doesn't work for me, I was hoping someone could manually verify my extension, thus I opened the issue on here. Sad to see that Flathub folks don't even consider that extensions should have their own pages, less so verification badges. As an extension maintainer, I find this upsetting. Extensions have been having a lot of issues in Flatpak & Flathub, as if they were added as an afterthought and treated as 3rd class citizens. (For example of the issues: extensions don't get updated to their new runtime version when the app updates its runtime, resulting in a user updating the app having all the installed extensions being dropped https://github.com/flatpak/flatpak/issues/4208, or, when installing an extension and flathub asks the user which of the numerous branches of the extension to install, flatpak does not indicate which branch will actually work with the app the user has installed and which will do nothing if installed https://github.com/flatpak/flatpak/issues/5581, and now these dedicated page and the verification badge issues). I can only hope that things get better for extensions over time.

**Comment by bbhtt** [comment]
> I don't care about owning the app-id of the extension, nor do I claim to be a VLC developer

That's how verification works. You confirm that you or the person publishing, own the domain used in the ID. That is what meant by "published by upstream developer" in the website in less
technical terms.

Also note that the docs say "app", an extension is not an "app", they are runtimes technically, and different from regular apps. An extension always has to follow the ID prefix of the extension point (in most cases this is the ID of the runtime/application) and majority of the extensions on Flathub are submitted by third parties with IDs not owned by them. So majority of them cannot verify, while some will be verified - which is ripe for immense confusion.

Also if we start manually verifying things it muddies the process and breaks user trust. It is supposed to be done by the developer/publisher without any interference.

Another issue, is combined with the ID `org.videlan` and the verification badge, it'll force people to think that it is published by VLC upstream, when it is not (someone can easily change the developer_name in metainfo to `VLC`). Then there is no way to verify extensions coming from upstream vs. extensions developed by third parties.

In short, the way flatpak extensions work and the verification system works aren't compatible and it's not your fault.

**Comment by hfiguiere** [comment]
There is no way to verify extensions. The badge should be removed instead.

See https://github.com/flathub/flathub/issues/5117
